.test-infra/jenkins/job_PreCommit_Java_Examples_Dataflow_Java11.groovy

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PrecommitJobBuilder
+import CommonJobProperties as commonJobProperties
+
+PrecommitJobBuilder builder = new PrecommitJobBuilder(
+        scope: this,
+        nameBase: 'Java_Examples_Dataflow_Java11',
+        gradleTask: ':clean', // Do nothing here
+        gradleSwitches: ['-PdisableSpotlessCheck=true'], // spotless checked in separate pre-commit
+        triggerPathPatterns: [
+                '^model/.*$',
+                '^sdks/java/.*$',
+                '^runners/google-cloud-dataflow-java/.*$',
+                '^examples/java/.*$',
+                '^examples/kotlin/.*$',
+                '^release/.*$',
+        ],
+        timeoutMins: 30,
+)
+builder.build {
+    publishers {
+        archiveJunit('**/build/test-results/**/*.xml')
+    }
+
+    steps {
+        gradle {
+            rootBuildScriptDir(commonJobProperties.checkoutDir)
+            tasks 'javaExamplesDataflowPrecommit'
+            switches "-Dorg.gradle.java.home=${commonJobProperties.JAVA_8_HOME}"
+            switches '-PdisableSpotlessCheck=true'
+            switches '-x runLegacyWorkerPreCommitTest'
+            switches '-x runApplianceTest'
+            switches '-x runWindmillTest'
+            commonJobProperties.setGradleSwitches(delegate)
+        }
+
+        gradle {
+            rootBuildScriptDir(commonJobProperties.checkoutDir)
+            tasks ':runners:google-cloud-dataflow-java:examples:runLegacyWorkerPreCommitTest'
+            tasks ':runners:google-cloud-dataflow-java:examples-streaming:runApplianceTest'
+            tasks ':runners:google-cloud-dataflow-java:examples-streaming:runWindmillTest'
+            switches '-x shadowJar'
```

Comment:
Doing these exclusions seems a fragile. Is there a way to indicate that some dependencies are provided from published cache and not built?

Reply:
@kennknowles thanks for all comments, first I thought I will answer to all your comments but then I decided to gather it all in just one. I can see there is lot of knowledge that is missing to me right now, can we contact outside this pr scope? (ex email?) I would very appreciate if you will share some knowledge with me. What do you think?